pkg/controllers/customroute/custom_route_controller.go

```diff
@@ -0,0 +1,509 @@
+
+	// this guarantees that route.Spec.Host is set to the current canonical host
+	if *modified || !equality.Semantic.DeepEqual(existingCopy.Spec, expectedRoute.Spec) {
+		// be careful not to print route.spec as it many contain secrets
+		existingCopy.Spec = expectedRoute.Spec
+		route, err = c.routeClient.Update(ctx, existingCopy, metav1.UpdateOptions{})
+		if err != nil {
+			return err
+		}
+	}
+
+	_, _, err = routeapihelpers.IngressURI(route, expectedRoute.Spec.Host)
+
+	return err
+}
+
+func (c *customRouteController) updateIngressConfigStatus(ctx context.Context, ingress *configv1.Ingress, customRouteErrors []error) error {
+	// update ingressConfig status
+	existingRoute, err := c.routeLister.Routes("openshift-authentication").Get("oauth-openshift")
+	if err != nil {
+		return err
+	}
+
+	_, err = c.updateComponentRouteStatus(ctx, &configv1.ComponentRouteStatus{
+		Namespace:        c.componentRoute.Namespace,
+		Name:             c.componentRoute.Name,
+		DefaultHostname:  configv1.Hostname("oauth-openshift." + ingress.Spec.Domain),
+		CurrentHostnames: []configv1.Hostname{configv1.Hostname(existingRoute.Spec.Host)},
+		ConsumingUsers: []configv1.ConsumingUser{
+			"system:serviceaccount:oauth-openshift:authentication-operator",
+		},
+		RelatedObjects: []configv1.ObjectReference{
+			{
+				Group:     routev1.GroupName,
+				Resource:  "routes",
+				Name:      "oauth-openshift",
+				Namespace: "openshift-authentication"},
+		},
+		Conditions: c.getConditions(customRouteErrors, existingRoute, ingress),
+	})
+	return err
+}
+
+func (c *customRouteController) getConditions(errs []error, route *routev1.Route, ingress *configv1.Ingress) []metav1.Condition {
+	now := metav1.Now()
+	if len(errs) != 0 {
+		return []metav1.Condition{
+			{
+				LastTransitionTime: now,
+				Type:               "Degraded",
+				Status:             metav1.ConditionTrue,
+				Reason:             "CustomRouteError",
+				Message:            fmt.Sprintf("Error Configuring custom route: %v", errs),
+			},
+			{
+				LastTransitionTime: now,
+				Type:               "Progressing",
+				Status:             metav1.ConditionFalse,
+				Reason:             "CustomRouteError",
+				Message:            fmt.Sprintf("Error Configuring custom route: %v", errs),
+			},
+		}
+	}
+
+	if _, _, err := routeapihelpers.IngressURI(route, route.Spec.Host); err != nil {
+		return []metav1.Condition{
+			{
+				LastTransitionTime: now,
+				Type:               "Degraded",
+				Status:             metav1.ConditionTrue,
+				Reason:             "RouteNotAdmitted",
+				Message:            fmt.Sprintf("Route not admitted: %v", err),
+			},
+			{
+				LastTransitionTime: now,
+				Type:               "Progressing",
+				Status:             metav1.ConditionFalse,
+				Reason:             "RouteNotAdmitted",
+				Message:            fmt.Sprintf("Route not admitted: %v", err),
+			},
+		}
+	}
+
+	if err := c.checkRouteAvailable(route.Spec.Host, ingress); err != nil {
```

Comment:
I would just push this outside in the sync once you've got a validated secret and then use the secret inside this function

Reply:
Right now this code only validates custom serving secrets, it would not handle cases where the default cert secret is in use.